src/topology/separation.lean

```diff
@@ -154,29 +154,31 @@ lemma union_right {a b c : set α} (ab : separated a b) (ac : separated a c) :
 
 end separated
 
-/-- A T₀ space, also known as a Kolmogorov space, is a topological space
-  where for every pair `x ≠ y`, there is an open set containing one but not the other. -/
+/-- A T₀ space, also known as a Kolmogorov space, is a topological space for every pair `x ≠ y`,
+there is an open set containing one but not the other. We formulate the definition in terms of
+the `inseparable` relation.  -/
 class t0_space (α : Type u) [topological_space α] : Prop :=
-(t0 : ∀ x y, x ≠ y → ∃ U:set α, is_open U ∧ (xor (x ∈ U) (y ∈ U)))
+(t0 : ∀ ⦃x y : α⦄, inseparable x y → x = y)
 
-lemma exists_is_open_xor_mem [t0_space α] {x y : α} (h : x ≠ y) :
-  ∃ U:set α, is_open U ∧ xor (x ∈ U) (y ∈ U) :=
-t0_space.t0 x y h
-
-lemma t0_space_def (α : Type u) [topological_space α] :
-  t0_space α ↔ ∀ x y, x ≠ y → ∃ U:set α, is_open U ∧ (xor (x ∈ U) (y ∈ U)) :=
-by { split, apply @t0_space.t0, apply t0_space.mk }
+lemma t0_space_iff_inseparable (α : Type u) [topological_space α] :
+  t0_space α ↔ ∀ (x y : α), inseparable x y → x = y :=
+⟨λ ⟨h⟩, h, λ h, ⟨h⟩⟩
 
 lemma t0_space_iff_not_inseparable (α : Type u) [topological_space α] :
   t0_space α ↔ ∀ (x y : α), x ≠ y → ¬inseparable x y :=
-by simp only [t0_space_def, xor_iff_not_iff, not_forall, exists_prop, inseparable_iff_forall_open]
+by simp only [t0_space_iff_inseparable, ne.def, not_imp_not]
 
-lemma t0_space_iff_inseparable (α : Type u) [topological_space α] :
-  t0_space α ↔ ∀ (x y : α), inseparable x y → x = y :=
-by simp only [t0_space_iff_not_inseparable, ne.def, not_imp_not]
+lemma t0_space_iff_exists_is_open_xor_mem (α : Type u) [topological_space α] :
+  t0_space α ↔ ∀ x y, x ≠ y → ∃ U:set α, is_open U ∧ (xor (x ∈ U) (y ∈ U)) :=
+by simp only [t0_space_iff_not_inseparable, xor_iff_not_iff, not_forall, exists_prop,
+  inseparable_iff_forall_open]
+
+lemma exists_is_open_xor_mem [t0_space α] {x y : α} (h : x ≠ y) :
+  ∃ U : set α, is_open U ∧ xor (x ∈ U) (y ∈ U) :=
+(t0_space_iff_exists_is_open_xor_mem α).1 ‹_› x y h
 
 lemma inseparable.eq [t0_space α] {x y : α} (h : inseparable x y) : x = y :=
```

Comment:
I would move this upwards, since this is now pretty much the definition of `t0_space`.